tests/rados/test_four_node_ecpool.py

```diff
@@ -0,0 +1,218 @@
+            f"with bulk flag enabled"
+            f"Expected : {pg_count_bulk_true}"
+        )
+    log.info(
+        "PGs increased to desired levels after application of bulk flag on the pool with no inactive PGs"
+    )
+
+    # Restarting OSDs belonging to a particular host
+    osd_node = ceph_cluster.get_nodes(role="osd")[0]
+    osd_list = rados_obj.collect_osd_daemon_ids(osd_node=osd_node)
+
+    for osd_id in osd_list:
+        log.debug(f"Rebooting OSD : {osd_id} and checking health status")
+        if not rados_obj.change_osd_state(action="restart", target=osd_id):
+            log.error(f"Unable to restart the OSD : {osd_id}")
+            raise Exception("Execution error")
+
+        time.sleep(5)
+        # Waiting for recovery to post OSD reboot
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj)
+        log.debug(
+            "PG's are active + clean post OSD reboot, proceeding to restart next OSD"
+        )
+
+    log.info(
+        f"All the planned  OSD reboots have completed for host {osd_node.hostname}"
+    )
+
+    # Beginning with OSD stop operations
+    log.debug("Stopping 1 OSD from each host. No inactive PGs")
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+    for node in osd_nodes:
+        osd_list = rados_obj.collect_osd_daemon_ids(osd_node=node)
+        if not rados_obj.change_osd_state(action="stop", target=osd_list[0]):
+            log.error(f"Unable to stop the OSD : {osd_list[0]}")
+            raise Exception("Execution error")
+        time.sleep(5)
+
+        pool_pgids = rados_obj.get_pgid(pool_name=pool_name)
+        for pgid in pool_pgids:
+            # Checking the PG state. There Should not be inactive state
+            pg_state = rados_obj.get_pg_state(pg_id=pgid)
+            if any("inactive" in key for key in pg_state.split("+")):
+                log.error(f"PG: {pgid} in inactive state during Pool bulk)")
+                raise Exception("PG Inactive. Test Fail")
+        log.debug("No inactive PGs found on the pool")
+
+        time.sleep(5)
+        # Waiting for recovery to post OSD reboot
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj)
+        log.debug(
+            "PG's are active + clean post OSD reboot, proceeding to restart next OSD"
+        )
+        if not rados_obj.change_osd_state(action="start", target=osd_list[0]):
+            log.error(f"Unable to stop the OSD : {osd_list[0]}")
+            raise Exception("Execution error")
+        time.sleep(5)
+
+    # Stopping all OSDs of 1 host and check for inactive PGs
+    stop_host = osd_nodes[0]
+    osd_list = rados_obj.collect_osd_daemon_ids(osd_node=stop_host)
+    for osd in osd_list:
+        if not rados_obj.change_osd_state(action="stop", target=osd):
+            log.error(f"Unable to stop the OSD : {osd_list[0]}")
+            raise Exception("Execution error")
+        time.sleep(5)
+
+        pool_pgids = rados_obj.get_pgid(pool_name=pool_name)
```

Comment:
Please consider moving the check for inactive PGs to a common module. Can be taken up later.

Reply:
Done.